agent/wsclient/client.go

```diff
@@ -161,14 +182,22 @@ func (cs *ClientServerImpl) Connect() error {
 		seelog.Warnf("Error creating a websocket client: %v", err)
 		return fmt.Errorf(string(resp) + ", " + err.Error())
 	}
-	cs.Conn = websocketConn
+	cs.conn = websocketConn
 	return nil
 }
 
+func (cs *ClientServerImpl) IsConnected() bool {
+	return cs.conn != nil
```

Comment:
Good catch @richardpen. I'd prefer to call it `IsReady()` as `IsNil()` might be leaking implementation detail.